src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_BasePatternSwitchLocalRewriter.cs

```diff
@@ -199,12 +205,14 @@ public override BoundNode VisitDynamicObjectCreationExpression(BoundDynamicObjec
 
                 public override BoundNode VisitObjectInitializerMember(BoundObjectInitializerMember node)
                 {
+                    // Although ref indexers are not declarable in C#, they may be usable
                     _mightAssignSomething = !node.ArgumentRefKindsOpt.IsDefault;
                     return base.VisitObjectInitializerMember(node);
                 }
 
                 public override BoundNode VisitIndexerAccess(BoundIndexerAccess node)
                 {
+                    // Although property arguments with ref indexers are not declarable in C#, they may be usable
                     _mightAssignSomething = !node.ArgumentRefKindsOpt.IsDefault;
```

Comment:
>_mightAssignSomething = !node.ArgumentRefKindsOpt.IsDefault; [](start = 20, length = 60)

Could this reset ```true``` to ```false```?

Reply:
`WhenClauseMightAssignWalker` now overrides `Visit` to short circuit any further visiting after it finds an assignment, so we will not get here when the value is `true`.

Reply:
> WhenClauseMightAssignWalker now overrides Visit to short circuit any further visiting after it finds an assignment, so we will not get here when the value is true.

Are we certain that none of the specific Visit... methods are called directly, or won't be called in the future? I think it is fine to use this approach as an optimization, but but it doesn't look sufficient to enforce correctness. 
Also, please add tests demonstrating that the problem is addressed.   

---
In reply to: [235163582](https://github.com/dotnet/roslyn/pull/31213#discussion_r235163582) [](ancestors = 235163582)